### ISSUE TITLE
Support mixed Objective-C and Swift projects

### ISSUE THREAD

**Issue by FelixSFD** [realm/jazzy#447]
Im running jazzy (0.5.0) with the following command, but only my Swift-files are parsed. All Objective-C headers are ignored.

> jazzy --clean --objc --umbrella-header MyApp/MyApp-Bridging-Header.h --framework-root /Applications/Xcode.app/Contents/Developer/Platforms/iPhoneOS.platform/Developer/SDKs/iPhoneOS9.1.sdk --author "The Author" --sdk iphoneos --swift-version 2.1 --min-acl private --xcodebuild-arguments -scheme,MyAppScheme --module MyApp

What am I doing wrong?

**Comment by khoogheem** [comment]
try it with
~~~
--framework-root . (assuming you are running jazzy in the root of your project)
~~~

Also, you shouldn't need --min-acl as that is for swift only I believe

**Comment by FelixSFD** [comment]
If the root ist the folder, where all my source-files are ([folder where the .xcodeproj is located]/MyApp/, I'm in root.

But it didn't work. Still parsing my Swift files only.

I have Swift and Objective-C in my project, so I need it.

**Comment by jpsim** [comment]
Sorry @FelixSFD, right now jazzy just documents *either* Objective-C or Swift projects. Generating documentation for Objective-C *and* Swift in the same project hasn't been added yet.

To my surprise, we didn't already have a ticket tracking that, so we're now using this one to track adding mixed documentation support.

**Comment by eflath** [comment]
As a workaround, is it possible to run two different commands and create separate documentation for iOS and swift? the command I'm running now documents the Swift code only even though i'm passing the --objc flag and I can't get it to parse the objc files

**Comment by jpsim** [comment]
Yes, you could combine the [SourceKitten](https://github.com/jpsim/SourceKitten) doc output for both Objective-C and Swift versions of your documentation using the `jazzy --sourcekitten-sourcefile`, but you're in *very* undocumented territory here so my ability to help you along will be limited.

**Comment by ronak2121** [comment]
Any update on this? It's been almost 8 - 9 months on it now.

**Comment by jpsim** [comment]
Nothing new to share, I don't think anyone's actively working on this. Would you like to contribute to this @ronak2121?

**Comment by jpsim** [comment]
still no

**Comment by wilmarvh** [comment]
Gonna take a guess here that there's no update yet @jpsim ?

If I were interested in contributing, where is the best place to start with this?

**Comment by johnfairh** [comment]
There are quite a few things to do to support mixing ObjC and Swift docs in the same website.
Broadly:
1. Add CLI flags etc. sufficient for jazzy to get two lots of sourcekitten output, one for objc and the other for Swift.
2. Work through all of jazzy's processing of sourcekitten output and figure out what needs doing, slowly getting rid of refs to the global 'objc' flag.  Figure out 'extensions of ObjC classes' and 'objc categories of swift classes'.
3. Figure out what the web pages ought to look like, what changes are needed there.

I would start with part 2, generating the sourcekitten output manually and passing it in with -s.  This comes in to `SourceKitten.parse` in `sourcekitten.rb` - would probably start off by setting an "objc or swift" flag in all the decls based off the filenames, then just follow all the code through and work through cases as they come up.

**Comment by johnfairh** [comment]
Fixed in master via #1113.
